src/figuresClasses.ts

```diff
@@ -1,11 +1,120 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  lue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+abstract class FigureBase {
+  static checkWrongSidesOfFigure(figureSidesArray: number[] | number): void {
+    if (typeof figureSidesArray === 'number') {
+      if (figureSidesArray <= 0) {
+        throw new Error('Radius is wrong (radius is <= 0)');
+      }
+    }
+
+    // we have to do Array.isArray(figureSidesArray),
+    // otherwise figureSidesArray.some((side: number) => side <= 0) does't work
+    if (
+      Array.isArray(figureSidesArray) &&
+      figureSidesArray.some((side: number) => side <= 0)
+    ) {
+      throw new Error('Length of some side is <= 0');
+    }
+  }
+}
+
+export class Triangle extends FigureBase implements Figure {
+  readonly shape: Shape = Shape.Triangle;
+
+  protected sidesArray: number[] = [this.a, this.b, this.c];
+
+  protected longestSide: number = [...this.sidesArray].splice(
+    this.sidesArray.indexOf(Math.max(...this.sidesArray)),
+  )[0];
+
+  protected otherSidesSum: number = this.sidesArray.reduce(
+    (acc: number, sideLength: number) => acc + sideLength,
+    0,
+  );
 
-export class Triangle implements Figure {}
+  protected semiPerimeter: number = (this.longestSide + this.otherSidesSum) / 2;
 
-export class Circle implements Figure {}
+  protected isSidesWrong: boolean = this.longestSide >= this.otherSidesSum;
```

Comment:
These values are dynamic and should be calculated instead of setting them statically. For example, you can have methods to do these calculations

Reply:
These values are not dynamic, they are static for every other figure. And also I need to calculate and use it only 1 time.
By plan (according to the task) I don't have to call these vars anywhere else outside of the class (and because of that I don't need a separate method for every of that calcs), so I made it protected.
Also calculations are not so complex to move them into separate method (even static), if I do this I'll just made more code and it won't be more understandable.